src/fraction.ts

```diff
@@ -0,0 +1,79 @@
+import BN from "bn.js";
+
+export class Fraction {
```

Comment:
Do classes like this not already exist in a standard library somewhere? Was there something about the implementation here that is so different from others?

Reply:
It has to work with BN as numerator/denominator